app/models/user.rb

```diff
@@ -0,0 +1,68 @@
+class User < ApplicationRecord
+	
+	SPECIAL = "?<>',?[]}{=-)(*&^%$#`~{}!"
+	REGEX = /[#{SPECIAL.gsub(/./){|char| "\\#{char}"}}]/
+
+	validates :first_name, :last_name, presence: true 
+	validates :password, format: { with: /(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[^\w\s]).{8,}/}
+
+	validate :password_lower_case, :password_uppercase, :password_special_char, :password_contains_number
+
+	before_create :set_password, :block_from_invitation?
+
+	attr_writer :login
+
+	def login
+		@login || self.username || self.email
+	end
+
+	devise :invitable, :database_authenticatable, :registerable,
+	:recoverable, :rememberable, :validatable
+
+	enum role: { user: 0, admin: 1}
```

Comment:
1. Place enum hash in a constant and use that constant here.
2. String key value pairs in enum

Reply:
{ user: 0, admin: 1} check spaces from start and end